proj/src/main/webapp/script.js

```diff
@@ -27,11 +27,72 @@ if(btnLogin){
     });
 }
 
+
+// Adds user to an existing chat when given a reference to the place in the database
+function addUserToTag(reference){
+    var currentUID = firebase.auth().currentUser.uid
+    console.log("adding new user to chat room with uid: " + currentUID);
+    reference.push(currentUID);
+}
+
+// Creates a new chat given a tag and adds the current user as a member
+function createNewChatWithUser(tag){
+    console.log("creating new chat with tag: " + tag);
+    var newChat = {
+        "name" : tag,
+        "tag" : tag,
+    };
+    var currentReference = firebase.database().ref("/chat/" + tag);
+    var postKey = currentReference.push(newChat).key;
+    currentReference = firebase.database().ref("/chat/" + tag + "/" + postKey + "/users/");
+    addUserToTag(currentReference);
+}
+
+// Create or join chatroom
+function createOrJoinChat(currentTag){
+    var ref = firebase.database().ref("/chat/");
+    ref.once("value").then(function(snapshot){
+        // Checks to see if tag already exists in database
+        if(snapshot.hasChild(currentTag)){
+
+            // If tag already exists, navigate to users and add users if there is room
+            var query = firebase.database().ref("/chat/" + currentTag + "/").orderByKey();
+            query.once("value").then(function(snapshot){
+
+                snapshot.forEach(function(childSnapshot){
+
+                    var currentReference = childSnapshot.ref;
+                    if(childSnapshot.hasChild("users")){
+
+                        var userSnapshot = childSnapshot.child("users");
+                        var usersReference = userSnapshot.ref;
+                        if(userSnapshot.numChildren() < 200){
```

Comment:
I'm not familiar with the Firebase libraries, so I'm not sure what this is, but it looks suspicious.  Is it limiting chat room size to 200?  If so, the 200 should be placed in a constant near the top of the file.  Does it fail silently when the condition is not met?  That sounds undesirable.

Reply:
I think they were planning to have the person join the room if it is less than 200 people and create a new subroom if it is greater, but yes I agree this should be in a constant near the top.